src/objects.rs

```diff
@@ -71,14 +71,67 @@ impl Id<Network> for Network {
 }
 impl_codes!(Network);
 
+#[derive(Clone, Debug)]
+pub struct Rgb {
+    pub red: u8,
+    pub green: u8,
+    pub blue: u8,
+}
+
+impl ::serde::Serialize for Rgb {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ::serde::Serializer,
+    {
+        let color = format!("{:02X}{:02X}{:02X}", self.red, self.green, self.blue);
+        serializer.serialize_str(&color)
+    }
+}
+
+impl<'de> ::serde::Deserialize<'de> for Rgb {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: ::serde::Deserializer<'de>,
+    {
+        use serde::de::Error;
+
+        let color_hex = String::deserialize(deserializer)?;
+        let color_dec = u32::from_str_radix(&color_hex, 16).map_err(Error::custom)?;
```

Comment:
Just add a check that the string length is 6 to not wait for FFF to be white.

Reply:
added